SpinCore_pp/config_parser_fn.py

```diff
@@ -0,0 +1,91 @@
+from pyspecdata import *
+import configparser
+
+class configuration(object):
+    # this registers the type, the pretty case we want, the section, and
+    # whether or not we can assume a default value
+    registered_params = {
+        "amplitude":(float, "acq_params", None),
+        "deadtime_us":(float, "acq_params", None),
+        "tau_us":(float, "acq_params", None),
+        "deblank_us":(float, "acq_params", None),
+        "SW_kHz":(float, "acq_params", None),
+        "acq_time_ms":(float, "acq_params", None),
+        "carrierFreq_MHz":(float, "acq_params", None),
+        "startconstant":(float, "acq_params", None),
+        "stopconstant":(float, "acq_params", None),
+        "p90_us":(float, "acq_params", None),
+        "gamma_eff_MHz_G":(float, "acq_params", 0.00425),
+        "field_width":(float,"acq_params",6),
+        "concentration":(float, "sample_params", None),
+        "krho_cold":(float, "sample_params", None),
+        "krho_hot":(float, "sample_params", None),
+        "T1water_hot":(float, "sample_params", None),
+        "T1water_cold":(float, "sample_params", None),
+        "repetition_us":(float, "sample_params", None),
+        "guessed_MHz_to_GHz":(float,"sample_params", 1.51671), # the ppt value we use to determine our rf carrier frequency based on our microwave
+        "max_power":(float, "odnp_params", None),
+        "uw_dip_center_GHz":(float, "odnp_params", None),
+        "uw_dip_width_GHz":(float, "odnp_params", None),
+        "FIR_rep":(float, "odnp_params", None),
+        "adc_offset":(int, "acq_params", None),
+        "nScans":(int, "acq_params", 1),
+        "thermal_nScans":(int, "acq_params",1),
+        "nEchoes":(int, "acq_params", None),
+        "IR_steps":(int, "acq_params", None),
+        "power_steps":(int, "odnp_params", None ),
+        "num_T1s":(int, "odnp_params", None),
+        "odnp_counter":(int, "file_names", 0),
+        "echo_counter":(int, "file_names", 0),
+        "cpmg_counter":(int, "file_names", 0),
+        "IR_counter":(int, "file_names", 0),
+        "field_counter":(int,"file_names",0),
+        "date":(int, "file_names", None),
+        "chemical":(str, "file_names", None),
+        "type":(str, "file_names", None),
```

Comment:
add a fourth argument to the tuple that documents each parameter
doesn't have to be fancy, but should make sense
e.g. "num_T1s":(int, "acq_params", None, "number of T1s acquried during ODNP")
look below for uses of "registered params" and make sure it can deal with the list of 4 -- see example below